### ISSUE TITLE
PluginVersion 0.3 not working with Sonar 5.2 (jdk1.8.0_40/postgresql-9.3)

### ISSUE THREAD

**Issue by stefanrinderle** [stefanrinderle/softvis3d#11]
_From @guidowojke on November 13, 2015 14:49_

Hallo everybody,

simply putting the file sonar-softVis3D-plugin-0.3.jar into the ./sonarqube-5.2/extensions/plugins folder and restarting sonar with sonar.sh file produces the following error:

uname -a:  Linux 3.2.0-4-amd64 1 SMP Debian 3.2.63-2+deb7u1 x86_64 GNU/Linux

```
--> Wrapper Started as Daemon
Launching a JVM...
Wrapper (Version 3.2.3) http://wrapper.tanukisoftware.org
  Copyright 1999-2006 Tanuki Software, Inc.  All Rights Reserved.

2015.11.13 15:25:52 INFO  app[o.s.p.m.JavaProcessLauncher] Launch process[search]: /export/home/bsci/tools/jdk/jdk1.8.0_40/jre/bin/java -Djava.awt.headless=true -Xmx1G -Xms256m -Xss256k -Djava.net.preferIPv4Stack=true -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=75 -XX:+UseCMSInitiatingOccupancyOnly -XX:+HeapDumpOnOutOfMemoryError -Djava.io.tmpdir=/export/home/bsci/server/sonarqube/sonarqube-5.2/temp -cp ./lib/common/*:./lib/search/* org.sonar.search.SearchServer /tmp/sq-process6222138717536646989properties
2015.11.13 15:25:53 INFO   es[o.s.p.ProcessEntryPoint]  Starting search
2015.11.13 15:25:53 INFO   es[o.s.s.SearchSettings]  Elasticsearch listening on 127.0.0.1:9001
2015.11.13 15:25:53 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] version[1.7.2], pid[15850], build[e43676b/2015-09-14T09:49:53Z]
2015.11.13 15:25:53 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] initializing ...
2015.11.13 15:25:53 INFO   es[o.e.plugins]  [sonar-1447424752708] loaded [], sites []
2015.11.13 15:25:53 INFO   es[o.elasticsearch.env]  [sonar-1447424752708] using [1] data paths, mounts [[/export (/dev/drbd0)]], net usable_space [871gb], net total_space [899.5gb], types [xfs]
2015.11.13 15:25:54 WARN   es[o.e.bootstrap]  JNA not found. native methods will be disabled.
2015.11.13 15:25:55 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] initialized
2015.11.13 15:25:55 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] starting ...
2015.11.13 15:25:55 INFO   es[o.e.transport]  [sonar-1447424752708] bound_address {inet[/127.0.0.1:9001]}, publish_address {inet[/127.0.0.1:9001]}
2015.11.13 15:25:55 INFO   es[o.e.discovery]  [sonar-1447424752708] sonarqube/u5SDHQiES1uYcG3EmBhwTw
2015.11.13 15:25:58 INFO   es[o.e.cluster.service]  [sonar-1447424752708] new_master [sonar-1447424752708][u5SDHQiES1uYcG3EmBhwTw][bscibeadev02.be.server.lan][inet[/127.0.0.1:9001]]{rack_id=sonar-1447424752708}, reason: zen-disco-join (elected_as_master)
2015.11.13 15:25:58 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] started
2015.11.13 15:25:58 INFO   es[o.e.gateway]  [sonar-1447424752708] recovered [6] indices into cluster_state
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=160m; support was removed in 8.0
2015.11.13 15:25:59 INFO  app[o.s.p.m.Monitor] Process[search] is up
2015.11.13 15:25:59 INFO  app[o.s.p.m.JavaProcessLauncher] Launch process[web]: /export/home/bsci/tools/jdk/jdk1.8.0_40/jre/bin/java -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Djruby.management.enabled=false -Djruby.compile.invokedynamic=false -Xmx768m -Xms256m -XX:MaxPermSize=160m -XX:+HeapDumpOnOutOfMemoryError -Djava.net.preferIPv4Stack=true -Dhttps.proxyHost=********** -Dhttps.proxyPort=3128 -Djava.io.tmpdir=/export/home/bsci/server/sonarqube/sonarqube-5.2/temp -cp ./lib/common/*:./lib/server/*:/export/home/bsci/server/sonarqube/sonarqube-5.2/lib/jdbc/postgresql/postgresql-9.3-1102-jdbc41.jar org.sonar.server.app.WebServer /tmp/sq-process5056604064872068091properties
2015.11.13 15:26:00 INFO  web[o.s.p.ProcessEntryPoint] Starting web
2015.11.13 15:26:00 INFO  web[o.s.s.app.Webapp] Webapp directory: /export/home/bsci/server/sonarqube/sonarqube-5.2/web
2015.11.13 15:26:00 INFO  web[o.a.c.h.Http11NioProtocol] Initializing ProtocolHandler ["http-nio-0.0.0.0-8999"]
2015.11.13 15:26:00 INFO  web[o.a.t.u.n.NioSelectorPool] Using a shared selector for servlet write/read
2015.11.13 15:26:01 INFO  web[o.e.plugins] [sonar-1447424752708] loaded [], sites []
2015.11.13 15:26:01 INFO  web[o.s.s.p.ServerImpl] SonarQube Server / 5.2 / f045be98be3ba032b6b19d71574038eeeb91803b
2015.11.13 15:26:01 INFO  web[o.sonar.db.Database] Create JDBC data source for jdbc:postgresql://*************/sonar
2015.11.13 15:26:02 INFO  web[o.s.s.p.DefaultServerFileSystem] SonarQube home: /export/home/bsci/server/sonarqube/sonarqube-5.2
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin Build Stability / 1.3 / adf81f731fc54be04431d4ccf753900c44ec33bd
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin Checkstyle / 2.3 / 7c62fe8a00123b8df1fa16cc25ad77888e50e811
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin Findbugs / 3.3 / e1a76544ee6aed765106e7e2bb64072adf63ccaf
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin Git / 1.1 / 21e7329a632904350bb9a2e7f1b17b9967988db8
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin Java / 3.6 / 88ff47a7574edcc4472ff495c15b6f94d8dbd98f
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin LDAP / 1.4 / 18ba3a13ba4c6c07e6b0ad77940a52734159c27e
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin PHP / 2.6 / 153f397dc6cac76c93eca5df541278447ed85ea4
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin PMD / 2.4.1 / 2f30e0a7177041a4fb9f7ecefdb301e2bfc15903
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin SVN / 1.2 / d04c3cdb21f48905dd8300d1129ec90281aa6db2
2015.11.13 15:26:02 INFO  web[o.s.s.p.ServerPluginRepository] Deploy plugin SoftVis3D / 0.3 / c5f74dec1790d1b3945855d18f29757a6f2d29ba
2015.11.13 15:26:03 INFO  web[o.s.s.p.RailsAppsDeployer] Deploying Ruby on Rails applications
2015.11.13 15:26:03 INFO  web[o.s.s.p.UpdateCenterClient] Update center: http://update.sonarsource.org/update-center.properties (HTTP proxy: ***********:3128)
2015.11.13 15:26:04 ERROR web[o.a.c.c.C.[.[.[/sonar]] Exception sending context initialized event to listener instance of class org.sonar.server.platform.PlatformServletContextListener
java.lang.NoClassDefFoundError: org/sonar/api/database/DatabaseSession
	at java.lang.Class.getDeclaredConstructors0(Native Method) ~[na:1.8.0_40]
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671) ~[na:1.8.0_40]
...
2015.11.13 15:26:08 INFO  web[o.a.c.h.Http11NioProtocol] Pausing ProtocolHandler ["http-nio-0.0.0.0-8999"]
2015.11.13 15:26:09 INFO  web[o.a.c.h.Http11NioProtocol] Stopping ProtocolHandler ["http-nio-0.0.0.0-8999"]
2015.11.13 15:26:09 INFO  web[o.a.c.h.Http11NioProtocol] Destroying ProtocolHandler ["http-nio-0.0.0.0-8999"]
2015.11.13 15:26:09 INFO  web[o.s.s.a.TomcatAccessLog] Web server is stopped
2015.11.13 15:26:09 INFO  app[o.s.p.m.TerminatorThread] Process[search] is stopping
2015.11.13 15:26:09 INFO   es[o.s.p.StopWatcher]  Stopping process
2015.11.13 15:26:09 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] stopping ...
2015.11.13 15:26:09 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] stopped
2015.11.13 15:26:09 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] closing ...
2015.11.13 15:26:09 INFO   es[o.elasticsearch.node]  [sonar-1447424752708] closed
2015.11.13 15:26:09 INFO  app[o.s.p.m.TerminatorThread] Process[search] is stopped
<-- Wrapper Stopped
```

After removing the plugin and restarting again, everything is running fine again.

Regards,
Guido

_Copied from original issue: stefanrinderle/sonar-softvis3d-plugin#5_

**Comment by stefanrinderle** [comment]
Hey,

first of all, thanks for trying to use my plugin. Unfortunately, my plugin is not working with version 5.2 of SonarQube yet. As I've been very busy the last few weeks and SonarQube 5.2 was released 2 weeks ago I did not had the time to state that correctly on my project page or the README file. This is done now.

The database access is not longer available for plugins in version 5.2. This causes the ClassNotFoundException during the initialization. It would be great to check the version during the initialization to avoid SonarQube from being stopped, but I don't know a way to do that.

It would be great to get an alternative or workaround for the server side code of the plugin to gather any data. It is also not possible to access the web service interface from the server side somehow. As a result, I have no data access currently. There's an open discussion on the sonarqube [google group](https://groups.google.com/forum/#!searchin/sonarqube/Plugin$20development$20with$20SonarQube$20%3E$3D$205.2/sonarqube/Ly6CaKGJpHA/9CSirmQY5wEJ) about that topic. More details on that in [my blog](http://softvis3d.com/#/blog).

Long story short. This issue can only be fixed if the platform changes, but I'm pretty confident that they will provide a way.

**Comment by stefanrinderle** [comment]
Form version 0.4, the Plugin Workshop again with the current SQ versions >= 5.5.
